R/iterate-helpers.R

```diff
@@ -0,0 +1,115 @@
+#' Iteration helpers
+#'
+#' @description
+#' These functions are suitable for use with the `next_req` argument to
+#' [req_perform_iteratively()]. Each implements iteration for a common
+#' pagination pattern:
+#'
+#' * `iterate_with_offset()` increments a query parameter, e.g. `?page=1`,
+#'   `?page=2`, or `?offset=1`, `offset=21`.
+#' * `iterate_with_cursor()` updates a query parameter with the value of a
+#'    cursor found somewhere in the response.
+#'  * `iterate_with_link_url()` follows the url found the `Link` header.
+#'    See `resp_link_url()` for more details.
+#'
+#' @param param_name Name of query parameter.
+#' @param start Starting value.
+#' @param offset Offset for each page.
```

Comment:
Maybe explain that the default 1 works if paginating by page index and otherwise this should be the page size.